### PR TITLE
ENABLE-3598 Use Goreleaser to release Lilly CLI artifacts, and refactor GHA workflow to separate out CLI build + release pipeline

### PR DESCRIPTION
**Pull request by regangunther-linz**
Have already tested that it produced releases as expected and have deleted the test release to tidy things up. Have showed the test release on a fork of the repo here: https://github.com/regangunther-linz/landonline-openapi-linter-lilly/releases

### CODE REVIEW

File: .github/workflows/build-cli.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Zally CLI artifact build and release
+
+on:
+  push:
+    branches: [ main ]
+    tags:
```
Comment (bharathcp) [inline .github/workflows/build-cli.yml:6]: do we need this?

File: .github/workflows/build-cli.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      - '*'
+    paths:
+      - 'cli/**'
+  pull_request:
+    branches: [ main ]
```
Comment (bharathcp) [inline .github/workflows/build-cli.yml:11]: do u not need path qualifier for this trigger?

Comment (regangunther-linz) [reply .github/workflows/build-cli.yml:8]: Added now

File: .github/workflows/build-cli.yml (hunk 1 above)
Comment (regangunther-linz) [reply .github/workflows/build-cli.yml:5]: updated now, and updated tag filter in the step condition below

**Review by bharathcp — APPROVED** [review]

**Pull request merged**